sdk/appconfiguration/app-configuration/src/policies/throttlingRetryPolicy.ts

```diff
@@ -63,16 +128,14 @@ export class ThrottlingRetryPolicy extends BaseRequestPolicy {
           throw err;
         }
 
-        const delayBetweenRechecksInMs = 1;
-        await checkWithTimeout(
-          () => httpRequest.abortSignal?.aborted === true,
-          delayBetweenRechecksInMs,
-          delayInMs
-        );
-        if (httpRequest.abortSignal?.aborted) {
-          throw new AbortError("The operation was aborted.");
-        }
-        return await this.sendRequest(httpRequest.clone());
+        return await waitForTimeoutOrAbortOrResolve({
+          timeoutMs: delayInMs,
+          abortSignal: httpRequest.abortSignal,
+          actionFn: async () => {
+            return await this.sendRequest(httpRequest.clone());
+          },
+          timeoutMessage: `ServiceBusy: Unable to fulfill the request in ${delayInMs}ms when retried.`
```

Comment:
I would remove `ServiceBusy: ` here - there actually is a real HTTP status code for service busy and you don't want to cause any confusion with people thinking that's what happening here.